imap_processing/cdf/global_attrs.py

```diff
@@ -0,0 +1,353 @@
+    fieldname: str, default=None
+        The fieldname, "FIELDNAM" attribute
+    var_type: str, default="support_data"
+        The type of data
+    fill_val: np.int64, default=Constants.INT_FILLVAL
+        The values for filling data
+    scale_type: str, default="linear"
+        The scale of the axis, "SCALETYP" attribute
+    label_axis: str, default=None
+        Axis label, "LABLAXIS" attribute
+    format: str, default=None
+        The format of the data, in Fortran format
+    units: str, default=None
+        The units of the data
+    """
+
+    validmin: Union[np.float64, np.int64]
+    validmax: Union[np.float64, np.int64]
+    display_type: str = None
+    catdesc: str = None
+    fieldname: str = None
+    var_type: str = "support_data"
+    fill_val: np.int64 = GlobalConstants.INT_FILLVAL
+    scale_type: str = "linear"
+    label_axis: str = None
+    format: str = None
+    units: str = ""
+
+    def output(self):
+        """
+        Generate the output for the data level as a dictionary.
+
+        Returns
+        -------
+        dict
+            Dictionary of correctly formatted values for the attributes in the class
+        """
+        return {
+            "CATDESC": self.catdesc,
+            "DISPLAY_TYPE": self.display_type,
+            "FIELDNAM": self.fieldname,
+            "FILLVAL": self.fill_val,
+            "FORMAT": self.format,
+            "LABLAXIS": self.label_axis,
+            "UNITS": self.units,
+            "VALIDMIN": self.validmin,
+            "VALIDMAX": self.validmax,
+            "VAR_TYPE": self.var_type,
+            "SCALETYP": self.scale_type,
+        }
+
+
+@dataclass
+class ScienceAttrs(AttrBase):
+    """
+    The class for Science attributes, in particular with depend_0 attributes.
+
+    It also contains all the attributes and defaults from the generic TypeBase as well.
+
+    Attributes
+    ----------
+    depend_0: str = None
+        The first degree of dependent coordinate variables.
+        Although this is an optional keyword, it is required for every instance.
+    depend_1: str = None, optional
+        The second degree of dependent coordinate variables. This is used for 2d data.
+    depend_2: str = None, optional
+        The third degree of dependent coordinate variables. This is used for 3d data.
+        If this variable is used, there must also be a depend_1 value.
+    variable_purpose: str = None, optional
+        The variable purpose attribute tells which variables are worth plotting.
+    var_notes: str = None, optional
+        Notes on the variable
+    """
+
+    depend_0: str = None
+    depend_1: str = None
+    depend_2: str = None
+    variable_purpose: str = None
+    var_notes: str = None
+
+    def __post_init__(self):
+        """If depend_0 is not set, raise an error, as this attribute is required."""
+        if self.depend_0 is None:
+            raise TypeError("ScienceBase requires depend_0 attribute.")
```

Comment:
Sure, I can add a test on some of these classes just to catch errors like the one Bryan caught earlier, where the attribute was in the dataclass but not in the output. Since these aren't expected to have attributes removed I think that makes a certain amount of sense.